airflow/providers/apache/spark/decorators/pyspark_submit.py

```diff
@@ -0,0 +1,205 @@
+        python_callable: Callable,
+        op_args: Sequence | None = None,
+        op_kwargs: dict | None = None,
+        use_dill: bool = False,
+        expect_airflow: bool = False,
+        **kwargs,
+    ):
+        self.use_dill = use_dill
+        self.expect_airflow = expect_airflow
+
+        signature = inspect.signature(python_callable)
+        parameters = [
+            param.replace(default=None) if param.name in SPARK_CONTEXT_KEYS else param
+            for param in signature.parameters.values()
+        ]
+        # mypy does not understand __signature__ attribute
+        # see https://github.com/python/mypy/issues/12472
+        python_callable.__signature__ = signature.replace(parameters=parameters)  # type: ignore[attr-defined]
+
+        if kwargs.get("application"):
+            if not conf.getboolean("operators", "ALLOW_ILLEGAL_ARGUMENTS"):
+                raise AirflowException(
+                    "Invalid argument 'application' were passed to `@task.pyspark_submit`."
+                )
+            warnings.warn(
+                "Invalid argument 'application' were passed to @task.pyspark_submit.",
+                UserWarning,
+                stacklevel=2,
+            )
+        if kwargs.get("application_args"):
+            if not conf.getboolean("operators", "ALLOW_ILLEGAL_ARGUMENTS"):
+                raise AirflowException(
+                    "Invalid argument 'application_args' were passed to `@task.pyspark_submit`."
+                )
+            warnings.warn(
+                "Invalid argument 'application_args' were passed to `@task.pyspark_submit`.",
+                UserWarning,
+                stacklevel=2,
+            )
+        for key in SPARK_CONTEXT_KEYS:
+            if key in kwargs:
+                if not conf.getboolean("operators", "ALLOW_ILLEGAL_ARGUMENTS"):
+                    raise AirflowException(
+                        f"Invalid key '{key}' in op_kwargs. You don't need to set it because it's a "
+                        "variable that will be automatically set within the Python process of the Spark "
+                        "job submitted via spark-submit."
+                    )
+                warnings.warn(
+                    f"Invalid key '{key}' in op_kwargs. You don't need to set it because it's a "
+                    "variable that will be automatically set within the Python process of the Spark "
+                    "job submitted via spark-submit.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+
+        super().__init__(
+            python_callable=python_callable,
+            op_args=op_args,
+            op_kwargs=op_kwargs,
+            **kwargs,
+        )
+
+    def execute(self, context: Context):
+        with TemporaryDirectory() as tmp_dir:
+            script_filename = os.path.join(tmp_dir, "script.py")
+            input_filename = os.path.join(tmp_dir, "SCRIPT__GENERATED__AIRFLOW.IN")
```

Comment:
This pattern follows `_DockerDecoratedOperator`.

https://github.com/apache/airflow/blob/e827bfbd2c79f3432411e14351c245a06410766a/airflow/providers/docker/decorators/docker.py#L95-L115